[11.x] Makes dividing easier with the `group` method.

selcukcukur opened the pull request:
### `group` Method

This method divides a given number into pairs of specified intervals, returning each segment in an array. It also supports adding delimiters and padding characters.

#### Parameters :
- `$number` : The number to be masked. This parameter can be a string, integer, or float.
- `$interval` : The length of each segment. The default value is 2.
- `$delimiter` : The character used to separate segments. The default value is an empty string ('').
- `$padding` : The character used to pad the last segment if it is shorter. The default value is an empty string ('').

#### Return :
- `string` The masked number with segments joined by the specified delimiter.

#### Example :

```php
use Illuminate\Support\Number;

// Example 1
Number::group(4465646434546464, 4); // Output : 4465 6464 3454 6464

// Example 2
Number::group('4465646434546464', 4, '-'); // Output : 4465-6464-3454-6464

// Example 3
Number::group(446564643454, 4, '-', 3434); // Output : 4465-6464-3454-3434

// Example 4
Number::group(12345.678, 2, '-'); // Output : 12-34-5.678

// Example 5
Number::group(4465646434546464, 0); // Output : 4465646434546464

// Example 6
Number::group(4465646434546464, 2, ':'); // Output : 44:65:64:64:34:54:64:64
```

selcukcukur committed:
[11.x] Makes dividing easier with the `divides` method.

selcukcukur committed:
fix

PerryvanderMeer commented:
With `Number::divides` I am more likely to think of mathematical division anyway 🤔

shaedrich commented:
I would expect `Number::divides(4465646434546464, 4);` to equal `4465646434546464 / 4`, so maybe the naming could be improved:
* `Number::group`
* `Number::divideIntoGroups()`
* `Number::formatToGroups()`

rodrigopedra commented:
1. shouldn't this be a string helper?
  - This basically manipulates the input as a string of characters. I understand it is meant to format card numbers, etc., but nonetheless, still a string operation
3. as others have pointed out, `divides` is an odd name, split, chunk, or group, maybe?

shaedrich commented:
> 1. shouldn't this be a string helper?

We might need a separate `Format` helper at some point 😂

selcukcukur committed:
Code understandability and syntax improved.

It was made safer by using built-in helpers, and the name divides was renamed to group because it could have the wrong connotations.

selcukcukur committed:
fix

selcukcukur commented:
You are right, group name will be better. And some fixes for syntax were applied via Laravel built-in helper classes. It seems better now. 👍🏼 @shaedrich @rodrigopedra

selcukcukur committed:
Update Number.php

selcukcukur committed:
Update Number.php

selcukcukur committed:
fix

rodrigopedra commented on src/Illuminate/Support/Number.php:
```diff
@@ -24,6 +24,45 @@ class Number
+        $number = (string) $number;
+
+        // Handle decimal part separately
+        [$integer, $decimal] = explode('.', $number.'.0');
```
Not all numbers use a dot as a decimal separator. As `$number` accepts a string, that can be a user provided string from a form.

I'd either accept the separator as a parameter, defaulting to a dot, or drop the support for decimal places all together (check if `$number` is a float and truncate it).

shaedrich replied:
And if one would want to parse the number in a localized way, NumberFormatter would provide such functionality. One would refrain from doing it oneself.

rodrigopedra commented:
I like this addition, but I still think this should go into the `Illuminate\Support\Str` class instead.

Let's wait for the maintainers input on this.
